active/IndexPath/IndexPath.md

```diff
@@ -0,0 +1,96 @@
+
+![Sample tree selection](./sample-tree-selection.png)
+
+### Creating IndexPath objects
+New IndexPaths objects can be created using the static `IndexPath.CreateFrom` and `IndexPath.CreateFromIndices` methods:
+
+```csharp
+
+// Points to the third element in a flat collection
+var simplePath = IndexPath.CreateFrom(2);
+
+// This is the same IndexPath as "simplePath"
+var simplePathIndices = IndexPath.CreateFromIndices(new List<int>(){ 2 });
+
+// This will point to the third child of the second element in a nested collection
+var groupedPath = IndexPath.CreateFrom(1,2);
+
+// The following IndexPath is equivalent to "groupedPath"
+var groupedPathIndices = IndexPath.CreateFromIndices(new List<int>(){ 1, 2 });
+
+```
+
+### Comparing two IndexPaths
+Comparing two `IndexPath` objects can be done using the `CompareTo` function:
+
+```c#
+var baseIndexPath = IndexPath.CreateFromIndices(new List<int> { 1, 4, 2 });
+
+var shorterPath = IndexPath.CreateFrom(2, 0);
+
+// Since shorterPath takes second element of first nesting, it is "greater" than baseIndexPath
+Assert.AreEqual(1, shorterPath.CompareTo(baseIndexPath));
+
+
+var sameLengthPath = IndexPath.CreateFromIndices(new List<int> { 1, 0, 1 });
+
+// Since sameLength path enters before baseIndexPath on the second level, it is smaller than basePath
+Assert.AreEqual(-1, sameLengthPath.CompareTo(baseIndexPath));
+
+
+var baseIndexPathCopy = IndexPath.CreateFromIndices(new List<int> { 1, 4, 2 });
+// Those two are the same, and thus CompareTo returns 0
+Assert.AreEqual(0, baseIndexPathCopy.CompareTo(baseIndexPath));
+```
+
+# API Notes
+
+| Function | Description |
+|-|-|
+|Int32 GetSize()| Returns the number of nesting levels this IndexPath addresses. |
+|Int32 GetAt(Int32 index)| Returns the index for the item in a given nesting level. |
+| Int32 CompareTo(IndexPath other)| Compares the IndexPath to a different IndexPath. If the other IndexPath is behind the current IndexPath, the method returns -1, if the other IndexPath is before the current IndexPath it returns 1. In case of equality this method returns 0. |
+| static IndexPath CreateFrom(Int32 index) | Creates an IndexPath that only consists of the index provided. |
+| static IndexPath CreateFrom(Int32 groupIndex, Int32 itemIndex) | Creates an IndexPath with the given group index and item index for the given group. |
+| static IndexPath CreateFromIndices(IVector<Int32> indices) | Creates an IndexPath from the given list of integers.
+
+# API Details
+
+```c++
+[WUXC_VERSION_MUXONLY]
+[webhosthidden]
+runtimeclass IndexPath : Windows.Foundation.IStringable
+{
+    Int32 GetSize();
+    Int32 GetAt(Int32 index);
+    Int32 CompareTo(IndexPath other);
+
+    [default_overload] [method_name("CreateFrom")]
+    static IndexPath CreateFrom(Int32 index);
+    [method_name("CreateFromGroupAndItemIndex")]
+    static IndexPath CreateFrom(Int32 groupIndex, Int32 itemIndex);
```

Comment:
I don't think it's obvious enough that two parameters means the grouping case, it would be more clear with a name like CreateForGroup, or some such with "group" in its name.

But how is the grouped case different from the indices case?

Reply:
The only difference is that it's a bit faster and easier to use in the two index case. Updated the name to CreateFromGroupAndItemIndex.

Reply:
Yes. These are convenience constructors. The most common are one and two indices(one level grouping) in which case from a usage perspective we don't have to require creating a list of one or list of two items.